src/check.rs

```diff
@@ -63,7 +63,7 @@ pub fn is_available(url: &Url, ctx: &CheckContext) -> Result<(), CheckError> {
 fn check_file_url(url: &Url, _ctx: &CheckContext) -> Result<(), CheckError> {
     let path = url.to_file_path().unwrap();
 
-    if path.exists() {
+    if (path.exists() && path.is_file()) || (path.is_dir() && path.join("index.html").exists()) {
```

Comment:
The exists() check is now redundant. I think the `is_dir()` check can also be omitted since it's implied by the second cal.
```suggestion
    if path.is_file() || path.join("index.html").is_file() {
```